subtests/docker_cli/systemd/systemd.py

```diff
@@ -1,86 +1,155 @@
             self.sysd_action('stop', self.config['sysd_unit_file'])
             unlink(filepath)
             self.sysd_action('daemon-reload')
-        DockerImages(self).clean_all([self.config['name']])
+        DockerImages(self).clean_all([self.config['image_name']])
 
     @staticmethod
-    def sysd_action(action=None, sysd_unit=None):
+    def sysd_action(action, sysd_unit=None):
         command = 'systemctl {}'.format(action)
         if sysd_unit:
             command = '{} {}'.format(command, sysd_unit)
         utils.run(command, ignore_status=False)
 
-    @staticmethod
-    def read_socket(host, port):
-        max_read_tries = 10
-        num_read_try = max_read_tries
-        while num_read_try:
-            # using timeout 13-sec as it's 1-sec longer than two DNS timeouts
-            s = socket.create_connection((host, port), 13)
-            s.settimeout(2.0)
-            try:
-                time_from_socket = s.recv(512).strip()
-                if time_from_socket:
-                    s.shutdown(socket.SHUT_RDWR)
-                    return time_from_socket
-                num_read_try -= 1
-            except socket.timeout:
-                pass
-            time.sleep(1.0)
-        raise DockerTestError("Failed to read from socket with %d tries"
-                              % max_read_tries)
+    def sed_file(self, vars_dict):
+        tmpdirf = self.sub_stuff['tmpdirf']
+        tmp_write = '{}.tmp'.format(tmpdirf)
+        with open(tmpdirf, "r") as srcfile, open(tmp_write, "w") as dstfile:
+            dstfile.write(srcfile.read().format(**vars_dict))
+        rename(tmp_write, tmpdirf)
+
+# r"""
+# Summary
+# -------
+#
+# To test building an image using systemd unitfile
+#
+# Operational Summary
+# ---------------------
+#
+# #. Edit unitfile ``docker-build.service`` & copy it to /etc/systemd/system
+# #. Edit Dockerfile
+# #. Systemd service is started using this file to build an image.
+# #. Systemd service is stoped and image removed
+# """
+
+
+class systemd_build(systemd_base):
+
+    def initialize(self):
+        # update dict, edit unitfile & copy it to /etc/systemd/system/
+        super(systemd_build, self).initialize()
+        build_dict = {'dockerfiledir': self.sub_stuff['bindir'],
+                      'options': self.config['unit_opts']}
+        vars_dict = self.sub_stuff['vars_dict']
+        vars_dict.update(build_dict)
+        self.sed_file(vars_dict)
+        copyfile(self.sub_stuff['tmpdirf'], self.sub_stuff['unitf_dst'])
+        # copy Dockerfile to tmpdir, edit, then copy to bindir for build
+        self.sub_stuff['tmpdirf'] = join(self.sub_stuff['tmpdir'],
+                                         'Dockerfile')
+        tmpdirf = self.sub_stuff['tmpdirf']
+        self.sub_stuff['dkrfl_bk'] = '{}.bk'.format(tmpdirf)
+        for f in [tmpdirf, self.sub_stuff['dkrfl_bk']]:
+            copyfile(self.sub_stuff['dkrfl'], f)
+        vars_dict = {'img_from': self.config['img_frm']}
```

Comment:
Whoa. I'd like you to take a close look at this, possibly in conjunction with the subsequent sed_file. Please tell me what it's doing (what it's really doing, not what you think now that it's doing).

I'm sorry for not pointing it out directly, but this is a major bug and I think it's important for you to track it down on your own.